### PR TITLE
feat(spf): Add AirPlay support

### LINKED ISSUE #1801: Feature: Basic AirPlay Support for SPF
Similar to what was implemented for https://github.com/videojs/v10/issues/1260

The goal of this ticket is to add behavior-level support for AirPlay on SPF

### PR DESCRIPTION
**Pull request by spuppo-mux**
Closes #1801. Adds AirPlay support to the SPF HLS engine. 

## What it adds

- **`setupAirPlay` behavior**: on WebKit, appends a fallback `<source type="application/x-mpegURL">` so Safari can hand the native HLS stream to an AirPlay receiver while local playback runs through MSE. Composed into `createSimpleHlsEngine`.
- **Load suspension while casting**: a new `loadSuspended` state slot (the SPF analogue of hls.js `stopLoad()`) drives the segment-load FSM to a `'suspended'` state that halts the loaders, so we don't double-fetch alongside the receiver.
- **`disableRemotePlayback` sequencing + author intent (#1800)**: MMS needs the flag `true` to fire `sourceopen`; AirPlay needs it `false` to show the picker. These are sequenced, flip to `false` only once the MediaSource is open.  Author intent lives on its own adapter-written signal, so an explicit opt-out is honored and never confused with MMS's programmatic write. It's folded into the reactor's state derivation, so toggling it after attach() runs teardown/setup.

## Lifecycle correctness

- **Fallback source tracks the MediaSource.** The native-HLS <source> is appended only while `context.mediaSource` is open and removed when the slot clears, so on a src change it doesn't linger and get selected for local native HLS against the next MMS attach. Chosen over reacting to `sourceclose` (which fires on AirPlay engage, when the receiver still needs the source).
- **Actor teardown on MMS close**. When Safari closes the ManagedMediaSource (notably on an AirPlay handoff), the per-type buffer + segment-loader actors are torn down on `sourceclose`, so the in-flight append that stop deliberately leaves running doesn't keep calling appendBuffer on a defunct SourceBuffer and spamming `InvalidStateError`.

## Known limitation

After an AirPlay session ends, Safari closes our ManagedMediaSource and doesn't re-select the MSE `<source>`, so local MSE playback can't resume until the source is reloaded. Casting itself works; only the return-to-local is affected. The append errors this used to produce are now suppressed by tearing down the buffer actors on `sourceclose`. Tracked for a follow-up.

---

> [!NOTE]
> **Medium Risk**
> Touches MSE attach on Safari (ManagedMediaSource), playback loading FSM, and remote-playback flags; behavior is gated to WebKit and covered by tests, but cast/handoff edge cases (e.g. post-AirPlay local resume) remain a known limitation.
> 
> **Overview**
> Adds **AirPlay for the SPF HLS engine on WebKit** by bridging MSE local playback with a native HLS handoff path, and pausing segment loading while casting.
> 
> **ManagedMediaSource attach** no longer uses `srcObject`; it prepends an MSE `<source>` (blob URL) so a second native-HLS `<source>` from AirPlay setup can coexist per WebKit’s MSE+AirPlay pattern.
> 
> New **`setupAirPlay` behavior** (composed into `createSimpleHlsEngine`): on capable WebKit video, after `mediaSource` is open, appends `application/x-mpegURL` fallback from `presentation.url`, toggles `disableRemotePlayback` on the element only after open (MMS still needs `true` to open), honors author **`disableRemotePlayback`** on adapter state, and sets **`loadSuspended`** while `webkitCurrentPlaybackTargetIsWireless` is true.
> 
> **Load suspension**: `loadXSegments` gains a **`suspended`** state when `loadSuspended` is true; on entry it sends **`stop`** to video/audio/text segment loaders (abort pending queue, leave in-flight fetch alone). **`sourceclose`** on `mediaSource` tears down buffer/loader actors so appends don’t continue after Safari closes MMS on handoff.
> 
> **`SimpleHlsMedia`** exposes **`disableRemotePlayback`** IDL wired to engine state, separate from programmatic element flags.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e20a2d6b5570cb3867b01ec8a68431a319b0aa8c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/spf/src/playback/actors/dom/segment-loader.ts [hunk 1]
```diff
@@ -511,6 +527,11 @@ export function createSegmentLoaderActor(
               scheduleAll(allTasks, ctx);
             }
           },
+          stop: (_message, ctx) => {
+            const { runner } = ctx;
+            runner.abortPending();
+            ctx.transition('idle');
+          },
```
[the remark this answers is not in the document]
Comment (spuppo-mux) [reply packages/spf/src/playback/actors/dom/segment-loader.ts:534]: @cjpillsbury  I think this would collide with what you did in #1847. Lmk if you think it's worth addressing/tracking?

**Comment by spuppo-mux** [comment]
This work has been moved to https://github.com/videojs/v10/pull/1888
